docs/enhancements/support-node-labeling.md

```diff
@@ -0,0 +1,128 @@
+---
+title: node labeling support
+authors:
+- "@oamizur"
+  creation-date: 2022-12-20
+  last-updated: 2022-12-20
+---
+
+# Support node labeling during installation
+
+## Summary
+
+There was a requirement to support node roles during installation.  These roles 
+are extra roles, which are used by the user and not by any Openshift components.
+These labels should be applied on agent CR which represents a node in a spoke cluster.
+
+## Motivation
+
+Currently, nodes can be assigned with a single role which can be one of two values [ master, worker].
+The may want to add additional roles.  These roles are for the user usage only.
+
+### Goals
+- For ZTP flow: add capability to add and update labels on BMH which will be transferred 
+to agent CR.
+- For boot it yourself flow: add capability to add and update labels to agent CR after agent
+  has been created.
+
+### Non goals
+- Single way to set node labels before agent creation both for ZTP flow, and boot it yourself flow.
+- REST API support.  Node labeling will not be supported under REST API.
+
+## Proposal
+
+There are 2 implementation options: Using labels, or adding dedicated spec field.
+
+### Using labels
```

Comment:
because we are talking about a specific case of adding labels to nodes i don't think that we should use the labels on the agent for that.

Reply:
There is already field for node_labels, please just use it

Reply:
but if it will exist in the DB we will need to take into account that the controller will use that

Reply:
i assume you need to set them in cr and then in db as all other things we do, right? I just mean that field in db already exists and we need to set it in kubeapi flow

Reply:
Yes